### PR TITLE
Fix#6203: Refactor LineageRunner use

### PR DESCRIPTION
**Pull request by nahuelverdugo**
Fixes: #6203 #8364

### Describe your changes :
- Refactor the use of the LineageRunner library into a class that leads with all the calls to the library. (#8364)
- Skipped `COPY table FROM path` queries when doing lineage as agreed on #6203 .

### Type of change :
- [x] Bug fix
- [x] Improvement

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Ingestion: @open-metadata/ingestion

### CODE REVIEW

File: ingestion/src/metadata/ingestion/lineage/sql_lineage.py [hunk 1]
```diff
@@ -23,62 +22,30 @@
...
+        ]  # pylint: disable=unbalanced-tuple-unpacking
         if database_schema == "<default>":
             database_schema = None
-    return {"database": database, "database_schema": database_schema, "table": table}
+        return {
+            "database": database,
+            "database_schema": database_schema,
+            "table": table,
+        }
+    return {"database": database, "database_schema": database_schema, "table": None}
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/lineage/sql_lineage.py:48]: I believe we are not using this function anywhere!

File: ingestion/src/metadata/ingestion/lineage/parser.py [hunk 2]
```diff
@@ -37,214 +43,283 @@
...
 
-    return join_data
+        if insensitive_match(clean_query, ".*merge into .*when matched.*"):
+            clean_query = insensitive_replace(
+                raw_str=clean_query,
+                to_replace="when matched.*",  # merge into queries specific
+                replace_by="",  # remove it as LineageRunner is not able to perform the lineage
+            )
+
+        if insensitive_match(clean_query, "^COPY.*"):
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/lineage/parser.py:322]: is this the one for https://github.com/open-metadata/OpenMetadata/issues/6203? Could we add some info around the why in the comments itself pls?

File: ingestion/src/metadata/ingestion/lineage/parser.py [hunk 3]
```diff
@@ -37,214 +43,283 @@
...
+
+    @cached_property
+    def clean_table_list(self) -> List[str]:
+        """
+        Clean the table name if it has <default>.
+        :return: clean table names
+        """
+        return [get_formatted_entity_name(str(table)) for table in self.involved_tables]
+
+    @cached_property
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/lineage/parser.py:117]: as discussed, this is quite beautiful but not supported on 3.7, so we'll need to manually handle the caching around here https://docs.python.org/3/library/functools.html#functools.cached_property

Comment (nahuelverdugo) [reply ingestion/src/metadata/ingestion/lineage/parser.py:117]: Added a library to support it on Python 3.7

File: ingestion/src/metadata/ingestion/lineage/parser.py (hunk 2 above)
Comment (nahuelverdugo) [reply ingestion/src/metadata/ingestion/lineage/parser.py:324]: Comment added.

**Review by pmbrull — APPROVED** [review]

**Pull request merged**